src/ui/map.js

```diff
@@ -1409,6 +1409,25 @@ class Map extends Camera {
         return this._update();
     }
 
+    /**
+     * Removes feature state, setting it back to the default behavior. If only
+     * source is specified, removes all feature states of that source. If
+     * feature.id is also specified, removes all keys for that feature's state.
+     * If key is also specified, removes that key from that feature's state.
+     *
+     * @param {Object} feature Feature identifier. Feature objects returned from
+     * {@link Map#queryRenderedFeatures} or event handlers can be used as feature identifiers.
+     * @param {string | number} feature.id (optional) Unique id of the feature. Optional if key is not specified.
+     * @param {string} feature.source The Id of the vector source or GeoJSON source for the feature.
+     * @param {string} [feature.sourceLayer] (optional)  *For vector tile sources, the sourceLayer is
```

Comment:
If someone's using a GeoJSON source, and they provide a `sourceLayer` here, it'll break, won't it? (it would be an easy mistake to make, after all it looks like you're supposed to put something, and it's super easy to mix up "style layer" and "source layer").

Could we change the interface in some way to make it harder to make the mistake? Or alternatively, we could just change the code in SourceCache to _always_ hardwire a source layer of `_geojsonLayer` for geojson sources, even if another argument is provided?

Reply:
👍gonna make it throw an error on superfluous `sourceLayer`s for geojson. preferable to hardwiring, in case user meant to use a vector source